internal/cmdset/cmdset.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmdset
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/internal/util/setters"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/cmd/config/ext"
+	"sigs.k8s.io/kustomize/cmd/config/runner"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+)
+
+// NewSetRunner returns a command runner.
+func NewSetRunner(parent string) *SetRunner {
+	r := &SetRunner{}
+	c := &cobra.Command{
+		Use:     "set DIR --value [SETTER_NAME=SETTER_VALUE]",
+		Args:    cobra.MaximumNArgs(1),
+		PreRunE: r.preRunE,
+		RunE:    r.runE,
+	}
+	runner.FixDocs(parent, c)
+	r.Command = c
+	c.Flags().StringVar(&r.KeyValue, "value", "",
+		"optional flag, the value of the setter to be set to")
+	c.Flags().StringVar(&r.SetBy, "set-by", "",
+		"annotate the field with who set it")
+	c.Flags().StringVar(&r.Description, "description", "",
+		"annotate the field with a description of its value")
+	c.Flags().BoolVarP(&r.RecurseSubPackages, "recurse-subpackages", "R", false,
+		"sets recursively in all the nested subpackages")
+
+	return r
+}
+
+func SetCommand(parent string) *cobra.Command {
+	return NewSetRunner(parent).Command
+}
+
+type SetRunner struct {
+	Command            *cobra.Command
+	Set                setters.FieldSetter
+	SetBy              string
+	Description        string
+	KeyValue           string
+	Name               string
+	Value              string
+	ListValues         []string
+	RecurseSubPackages bool
+}
+
+func (r *SetRunner) preRunE(c *cobra.Command, args []string) error {
+	valueFlagSet := c.Flag("value").Changed
+
+	if !valueFlagSet {
+		return errors.Errorf("value flag must be set")
+	}
+
+	// r.KeyValue holds SETTER_NAME=SETTER_VALUE, split them into name and value
+	r.Name = strings.Split(r.KeyValue, "=")[0]
```

Comment:
Do we need to split it twice? And we should check the size of the slice before accessing the elements.

Reply:
Also, I seem to remember discussions around supporting multiple key-value pairs?

Reply:
Done. 

Yes. I want to start off simple and expand it further.